### PR TITLE
feat: migrate cpms client to github

### PR DESCRIPTION
**Pull request by faggc**
## Description

Migrate cpms/client module to github

Related issue: [BL-14412](https://dvsa.atlassian.net/browse/BL-14412)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  cancel-in-progress: true
+
+jobs:
+  static:
+    uses: dvsa/.github/.github/workflows/php-library-static.yml@v3.2.3
+
+  test:
+    uses: dvsa/.github/.github/workflows/php-library-tests.yml@v3.2.3
+    with:
+      php-versions: "[\"7.4\", \"8.0\", \"8.1\"]"
```
Comment (caoimheb) [inline .github/workflows/ci.yml]:
```suggestion
      php-versions: "[\"8.0\", \"8.1\"]"
```
v7 not needed here. `composer.json` for this repo uses at least v8.0

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  security:
+    uses: dvsa/.github/.github/workflows/php-library-security.yml@v3.2.3
+    with:
+     php-versions: "[\"7.4\", \"8.0\", \"8.1\"]"
```
Comment (caoimheb) [inline .github/workflows/ci.yml]:
```suggestion
     php-versions: "[\"8.0\", \"8.1\"]"
```

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!groovy
```
Comment (caoimheb) [inline Jenkinsfile]: This file can be deleted

File: README.md [hunk 4]
```diff
@@ -1,2 +1,166 @@
...
+* The generation of the required access token based on the scope 
+* Logging when a logger alias is provided
+* Caching of access tokens when caching is setup and enabled.
+ 
+
+Installation
+-------------
+The recommended way to install is through [Composer][composer].
+```
+composer require dvsa/authentication-ldap
```
Comment (caoimheb) [inline README.md]:
```suggestion
composer require dvsa/mot-cpms-client
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,166 @@
...
+                      'product_reference',
+                      'sales_reference',
+                      'payment_reference',
+                      'amount'
+                  ],
+              ]
+
+Contributing
+------------
+Please refer to our [Contribution Guide](/CONTRIBUTING.md) and [Contributor Code of Conduct](/CODE_OF_CONDUCT.md).
```
Comment (caoimheb) [inline README.md]:
```suggestion
Please refer to our [Contribution Guide](/CONTRIBUTING.md).
```
The code of conduct is linked through the contributing guide.

File: buildConfiguration.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (caoimheb) [inline buildConfiguration.yaml]: This file can be deleted

File: captainhook.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+            }
+        ]
+    },
+    "pre-commit": {
+        "enabled": true,
+        "actions": [
+            {
+                "action": "git secrets --pre_commit_hook --",
+                "options": []
+            }
```
Comment (caoimheb) [inline captainhook.json:24]: Can we add another action here to run the unit tests as part of pre-commit?

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
+{
+    "name": "cpms/client",
```
Comment (caoimheb) [inline composer.json]:
```suggestion
    "name": "dvsa/mot-cpms-client",
```

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    "type": "project",
+    "license": "proprietary",
+    "require": {
+        "php": "^8.0",
+        "laminas/laminas-mvc": "^3.3.0",
+        "laminas/laminas-cache": "^3.1.3",
+        "laminas/laminas-log": "^2.13",
+        "laminas/laminas-filter": "^2.13",
+        "laminas/laminas-dependency-plugin": "^2.2",
+        "dvsa/mot-cpms-notifications": "^1.0.0",
```
Comment (caoimheb) [inline composer.json]: I'd probably move this above the laminas dependencies just for tidiness

File: phpcs.xml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
```
Comment (caoimheb) [inline phpcs.xml:1]: Is the package-lock file above needed?

File: phpcs.xml [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="mot-cpms-client" xsi:noNamespaceSchemaLocation="./vendor-bin/php-codesniffer/vendor/squizlabs/php_codesniffer/phpcs.xsd">
+
+    <file>./src</file>
+    <file>./tests</file>
```
Comment (caoimheb) [inline phpcs.xml]:
```suggestion
    <file>./test</file>
```

File: phpstan.neon [hunk 12]
```diff
@@ -0,0 +1,7 @@
+parameters:
+    checkMissingIterableValueType: false
+    checkGenericClassInNonGenericObjectType: false
+    level: max
+    paths:
+        - src
+        - tests
```
Comment (caoimheb) [inline phpstan.neon]:
```suggestion
        - test
```

File: psalm.xml [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="https://getpsalm.org/schema/config"
+        xsi:schemaLocation="https://getpsalm.org/schema/config vendor-bin/psalm/vendor/vimeo/psalm/config.xsd"
+        xmlns:xi="http://www.w3.org/2001/XInclude"
+        resolveFromConfigFile="true"
+>
+
+    <projectFiles>
+        <directory name="src" />
+        <directory name="tests" />
```
Comment (caoimheb) [inline psalm.xml]:
```suggestion
        <directory name="test" />
```

File: vendor-bin/php-codesniffer/composer.json [hunk 14]
```diff
@@ -0,0 +1,10 @@
+{
+    "require": {
+        "php": "^7.4 || ^8.0",
```
Comment (caoimheb) [inline vendor-bin/php-codesniffer/composer.json]:
```suggestion
        "php": "^8.0",
```

File: vendor-bin/phpstan/composer.json [hunk 15]
```diff
@@ -0,0 +1,9 @@
+{
+    "require": {
+        "php": "^7.4 || ^8.0",
```
Comment (caoimheb) [inline vendor-bin/phpstan/composer.json]:
```suggestion
        "php": "^8.0",
```

File: vendor-bin/psalm/composer.json [hunk 16]
```diff
@@ -0,0 +1,9 @@
+{
+    "require": {
+        "php": "^7.4 || ^8.0",
```
Comment (caoimheb) [inline vendor-bin/psalm/composer.json]:
```suggestion
        "php": "^8.0",
```

File: captainhook.json (hunk 7 above)
Comment (faggc) [reply captainhook.json:24]: Added the action, had to add the script to composer first.

File: build.xml [hunk 17]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jennifer-choo) [inline build.xml]: Might want to remove this file and other build related files as I think they're related to the jenkins build

**Review by jennifer-choo — DISMISSED** [review]

File: build-scripts/clone-to-vhosts.sh [hunk 18]
```diff
@@ -0,0 +1,92 @@
+#!/bin/bash -x
```
Comment (DariuszJerzewski) [inline build-scripts/clone-to-vhosts.sh]: This can be removed.

File: phpunit.xml.dist [hunk 19]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (DariuszJerzewski) [inline phpunit.xml.dist]: I can't see the difference between this and phpunit.xml. Can this be removed?

Comment (faggc) [reply phpunit.xml.dist]: sure, within the repository phpunit.xml is a new file. For whatever reason it was originally distributed with this file. I imagine this is no longer necessary.

**Review by DariuszJerzewski — DISMISSED** [review]

File: CHANGELOG.md [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+## v1.24.2 - Wed Mar 23 2016
+
+### Fixes
+
+* Bumped minimal version of `cpms-notifications` to fix a bug in a third-party dependency
+
+## v1.24.1 - Thu Mar 17 2016
+
+### Testing
```
Comment (DariuszJerzewski) [inline CHANGELOG.md:33]: Can we also remove this section?

**Review by jennifer-choo — APPROVED** [review]

**Review by DariuszJerzewski — APPROVED** [review]

**Pull request merged**